test/Service/SessionManagerFactoryTest.php

```diff
@@ -84,12 +104,19 @@ public function testFactoryWillAddValidatorViaConfiguration()
                 'Zend\Session\Validator\RemoteAddr',
             ],
         ]];
-        $this->services->setService('Config', $config);
-        $manager = $this->services->get('Zend\Session\ManagerInterface');
+        $services = $this->services->withConfig([
+            'services' => [
+                'config' => $config,
+            ],
+        ]);
+        $manager = $services->get('Zend\Session\ManagerInterface');
 
         $manager->start();
 
-        $this->assertEquals(1, $manager->getValidatorChain()->getListeners('session.validate')->count());
+        $chain = $manager->getValidatorChain();
+        $r = new \ReflectionMethod($chain, 'getListenersByEventName');
```

Comment:
@Maks3w Correct. That's what the reflection calls are doing, essentially; verifying that the *factory* attached listeners for specific event names.